src/drivers/imx/esai.c

```diff
@@ -3,38 +3,450 @@
 // Copyright 2019 NXP
 //
 // Author: Daniel Baluta <daniel.baluta@nxp.com>
+// Author: Paul Olaru <paul.olaru@nxp.com>
 
+#include <sof/audio/component.h>
+#include <sof/drivers/edma.h>
 #include <sof/drivers/esai.h>
+#include <sof/lib/alloc.h>
 #include <sof/lib/dai.h>
 #include <sof/lib/dma.h>
 #include <ipc/dai.h>
+#include <ipc/topology.h>
+#include <errno.h>
+#include <stdint.h>
+
+struct esai_pdata {
+	struct {
+		uint32_t ecr;
+		uint32_t tfcr;
+		uint32_t rfcr;
+		uint32_t saicr;
+		uint32_t tcr;
+		uint32_t tccr;
+		uint32_t rcr;
+		uint32_t rccr;
+		uint32_t tsma;
+		uint32_t tsmb;
+		uint32_t rsma;
+		uint32_t rsmb;
+		uint32_t prrc;
+		uint32_t pcrc;
+	} regs;
+	/* TODO set the config here, prepare the registers and defer the
+	 * actual updating of the HW registers to the start() and stop()
+	 * triggers
+	 */
```

Comment:
Not sure it makes sense to cache the config here. Just write it as it is requested. This means remove this comment and keep the existing behaviour.